File: samples/server/petstore/finch/src/main/scala/Category.scala
package io.swagger.petstore

import argonaut.Argonaut._
import argonaut.CodecJson

/**
 * A category for a pet
 * @param id 
 * @param name 
 */
case class Category(id: Option[Long],
                name: Option[String]
                )

object Category {
  /**
   * Creates the codec for converting Category from and to JSON.
   */
  implicit val CategoryCodec: CodecJson[Category] = casecodec2(Category.apply, Category.unapply)("id", "name")
}

File: samples/client/petstore/flutter/swagger/lib/model/category.dart
part of swagger.api;

class Category {
  
  int id = null;
  

  String name = null;
  
  Category();

  @override
  String toString() {
    return 'Category[id=$id, name=$name, ]';
  }

  Category.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    id = json['id'];
    name = json['name'];
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name
     };
  }
}

